doc/merkle.md

```diff
@@ -0,0 +1,139 @@
+# Merkle Structures
+
+MimbleWimble is designed for users to verify the state of the system given
+only pruned data. To achieve this goal, all transaction data is committed
+to the blockchain by means of Merkle trees which should support efficient
+updates and serialization even when pruned.
+
+Also, almost all transaction data (inputs, outputs, excesses and excess
+proofs) have the ability to be summed in some way, so it makes sense to
+treat Merkle sum trees as the default option, and address the sums here.
+
+A design goal of Grin is that all structures be as easy to implement and
+as simple as possible. MimbleWimble introduces a lot of new cryptography
+so it should made as easy to understand as possible. Its validation rules
+are simple to specify (no scripts) and Grin is written in a language with
+very explicit semantics, so simplicity is also good to achieve well-understood
+consensus rules.
+
+## Merkle Trees
+
+There are three Merkle trees committed to by each block:
+
+### Total Output Set
+
+Each object is one of two things: a (commitment, rangeproof) pair indicating
+an unspent output or a NULL marker indicating a spent one. It is a sum-tree
+over all unspent outputs (spent ones contribute nothing to the sum).
```

Comment:
For the replay you describe it must be possible to create the same commitment again, right? I suggested to avoid that by requiring that every commitment has some digest of its input(s) worked in so it is not feasible to create same commitment that is not also a double spend. This would be similar to Bitcoin's replay protection. There reference to output is by transaction id + index where transaction id hashes the input references, therefore output could only be recreated if it were also a double spend.